Lib/test/test_netrc.py

```diff
@@ -123,6 +124,33 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
+            self.assertRaises(FileNotFoundError, netrc.netrc)
+
+    def test_file_not_found_explicit(self):
+        self.assertRaises(FileNotFoundError, netrc.netrc,
+                          file="unlikely_netrc")
+
+    def test_home_not_set(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        fake_netrc_path = os.path.join(d, 'netrc')
+        with open(fake_netrc_path, 'w') as f:
+            f.write("machine foo.domain.com login bar password pass")
+        os.chmod(fake_netrc_path, 0o600)
+        with support.EnvironmentVarGuard() as environ:
+            environ.unset('HOME')
+            with mock.patch('os.path.expanduser') as mock_expanduser:
+                mock_expanduser.return_value = fake_netrc_path
+                nrc = netrc.netrc()
```

Comment:
Check the content of nrc for testing that the configuration is read from the right file. Add a random data in the fake `.netrc`.